src/common/board.rs

```diff
@@ -56,14 +70,28 @@ pub trait Board: Status {
     fn get_digital_interrupt_value(&self, _pin: i32) -> anyhow::Result<u32> {
         anyhow::bail!("this board does not support digital interrupts")
     }
+    /// Get the pin's given duty cycle
     fn get_pwm_duty(&self, pin: i32) -> f64;
+    /// Set the pin to the given duty cycle 
     fn set_pwm_duty(&mut self, pin: i32, duty_cycle_pct: f64) -> anyhow::Result<()>;
+    /// Get the PWM frequency of the pin
     fn get_pwm_frequency(&self, pin: i32) -> anyhow::Result<u64>;
+    /// Set the pin to the given PWM frequency (in Hz). 
+    /// When frequency is 0, it will use the board’s default PWM frequency.
```

Comment:
This is not true, setting the frequency to 0 unregisters the pin and PWM channel from the timer and removes the PWM signal

Reply:
Definitely something that should be included. Thanks for clarifying.

Reply:
I looked at `esp32::board`'s implementation of this, and it doesn't unregister the PWM channel from the timer. It just sets the pin's `pwm_driver` to `None`. 

There is a `Drop` implementation for `PwmDriver` but I don't think it's ever actually dropped since `pin.pwm_driver` is `Option<PwmDriver<'static>>`. If so, we need to change how `0` is handled. Though this would be outside the scope of this PR.

Reply:
I've confirmed via print statement that it is in fact dropped